es5-shim.js

```diff
@@ -1980,9 +1980,15 @@
         parseInt = (function (origParseInt) {
             var hexRegex = /^[\-+]?0[xX]/;
             return function parseInt(str, radix) {
-                var string = trim(str);
-                var defaultedRadix = $Number(radix) || (hexRegex.test(string) ? 16 : 10);
-                return origParseInt(string, defaultedRadix);
```

Comment:
In firefox 17 parseInt(null) and parseInt(undefined) are throw "can't convert undefined to object".
But parseInt(NaN) is ok. Need some testing.

Probably that is because of 
`var trim = call.bind(String.prototype.trim);`